.github/ci.sh

```diff
@@ -33,6 +33,16 @@ extract_exe() {
   $IS_WIN || chmod +x "$2/$name"
 }
 
+extract_libffi() {
+  libffi=("$(ghc --print-libdir)"/rts/libffi.*)
+  mkdir -p "$1"
+  cp "${libffi[@]}" "$1"
+}
+
+extract_libffi_for_bin() {
+  extract_libffi "$(dirname "$(cabal v2-exec which "$1$EXT")")/../lib"
```

Comment:
One problem with putting the shared libraries under `../lib` is that we package up `dist/bin` as an artifact for use in later stages of the CI pipeline. (See [here](https://github.com/GaloisInc/cryptol/blob/9d68dc91ae7fe7fc5b7bb78484fe3909950afb25/.github/workflows/ci.yml#L206-L210).) As a result, this artifact will not include any of the shared libraries, since it doesn't include `dist/lib`. This doesn't break anything on CI currently, since none of these artifacts are used on OSes where the choice of `libffi` version matters, but this could cause problems down the road.

I think the simplest solution would be to copy these shared libraries to somewhere under `dist/bin` so that they are included in the artifacts.